auth.go

```diff
@@ -474,32 +435,87 @@ func (x *Central) Login(identity, password string) (sessionkey string, token *To
 }
 
 // Merges ldap with user store every 10 seconds
-func (x *Central) Merge() error {
+func (x *Central) MergeTicker() error {
 	x.Log.Info("Starting LDAP merge process")
-	x.Ticker = time.NewTicker(time.Second * 10)
+	x.Ticker = time.NewTicker(ldapMergeTickerSeconds)
 	go func() {
-		for _ = range x.Ticker.C {
+		mergeTick := func() error {
 			x.Log.Infof("Merging at (%v)", time.Now().UnixNano()/int64(time.Millisecond))
-			ldapUsers, err := x.userStore.GetLdapUsers()
-			//for _,ldapUser := range ldapUsers {
-			//	x.Log.Infof("User: %v", ldapUser.Username)
-			//}
+			ldapUsers, err := x.ldap.GetLdapUsers()
 			if err != nil {
-				x.Log.Errorf("Failed to retrieve users from LDAP server (%v)", err)
-				break
+				x.Log.Errorf("Failed to retrieve users from LDAP server for merge to take place (%v)", err)
+				return err
 			}
-			err = x.userStore.Merge(ldapUsers)
+			imqsUsers, err := x.userStore.GetIdentities()
 			if err != nil {
+				x.Log.Errorf("Failed to retrieve users from Userstore for merge to take place (%v)", err)
+				return err
+			}
+			if err = x.Merge(ldapUsers, imqsUsers); err != nil {
 				x.Log.Errorf("Failed to merge LDAP users (%v)", err)
-				break
+				return err
 			}
 			x.Log.Infof("Merging complete at (%v)", time.Now().UnixNano()/int64(time.Millisecond))
+			return nil
+		}
+		if err := mergeTick(); err == nil {
+			for _ = range x.Ticker.C {
+				if err := mergeTick(); err != nil {
+					break
+				}
+			}
+			x.Ticker.Stop()
+
 		}
-		x.Ticker.Stop()
 	}()
 
 	return nil
+}
+
+// Merges the ldap users and imqs users
+func (x *Central) Merge(ldapUsers []AuthUser, imqsUsers []AuthUser) error {
+	found := false
```

Comment:
`found` does not need to exist outside the scope of the `range ldapUsers` loop, and therefore it MUST be inside that loop. Having `found` be in the outer scope make it significantly harder to understand the code. Same applies to the Archive case.

I think it's important that we do a check here to avoid needless calls to UpdateIdentity. This allows us to log whenever a meaningful Update occurs. You'll want to create an "equals" function on `AuthUser` for this.